### PR TITLE
Hook 412 Implement NUMPAD from the new UI kit

### PR DESCRIPTION
**Pull request by VitalyKucher**
implementation

### CODE REVIEW

File: lib/widgets/text/numpad_text.dart [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+      } else if(identical(_dotKey, key)) {
+        if (!result.contains('.')) {
+          result += '.';
+        }
+      } else if(identical(_zeroKey, key)) {
+        result += '0';
+      } else if(identical(_clearKey, key)) {
+        result = result.substring(0, result.length-1);
+      }
+      if (result.contains('.') && result.substring(result.indexOf('.')).length > 7) {
```
Comment (almeynman) [inline lib/widgets/text/numpad_text.dart]: please add as a prop

File: lib/widgets/text/numpad_text.dart [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+  const NumPadText({
+    @required this.textEditingController
+  });
+
+  @override
+  _NumPadTextState createState() => _NumPadTextState();
+}
+
+class _NumPadTextState extends State<NumPadText> {
+  static const Key _sevenKey = Key('seven');
```
Comment (almeynman) [inline lib/widgets/text/numpad_text.dart]: check if keys are global, if yes then please change it to 'numpad-key-item-seven'

File: lib/widgets/text/numpad_text.dart [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+      if (result.contains('.') && result.substring(result.indexOf('.')).length > 7) {
+        result = result.substring(0, result.length-1);
+      }
+      widget.textEditingController.text = result;
+    });
+  }
+
+  KeyItem buildKeyItem(String val, Key key) {
+    return KeyItem(
+      key: key,
```
Comment (almeynman) [inline lib/widgets/text/numpad_text.dart]: value: '.'

File: lib/widgets/text/numpad_text.dart [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+  Widget build(BuildContext context) {
+    assert(debugCheckHasMaterial(context));
+    return Expanded(
+      child: Material(
+        type: MaterialType.transparency,
+        child: InkResponse(
+          radius: 30.0,
+          splashColor: AppColor.brightGreen,
+          highlightColor: Colors.white,
+          onTap: () => onKeyTap(key),
```
Comment (almeynman) [inline lib/widgets/text/numpad_text.dart]: onKeyTap(value)

File: lib/widgets/text/numpad_text.dart [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    );
+  }
+}
+
+class KeyItem extends StatelessWidget {
+
+  final Widget child;
+  @override
+  // ignore: overridden_fields
+  final Key key;
```
Comment (almeynman) [inline lib/widgets/text/numpad_text.dart]: final String value

File: lib/widgets/text/numpad_text.dart [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+  static const Key _threeKey = Key('three');
+  static const Key _dotKey = Key('dot');
+  static const Key _zeroKey = Key('zero');
+  static const Key _clearKey = Key('clear');
+
+  double height;
+  double width;
+  TextStyle digitTextStyle;
+
+  void onKeyTapped(Key key) {
```
Comment (almeynman) [inline lib/widgets/text/numpad_text.dart]: onKeyTapped(String value)

File: lib/widgets/text/numpad_text.dart [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+    var result = widget.textEditingController.text;
+    setState(() {
+      if (identical(_sevenKey, key)) {
```
Comment (almeynman) [inline lib/widgets/text/numpad_text.dart]: result += value

File: lib/widgets/text/numpad_text.dart (hunk 2 above)
Comment (almeynman) [inline lib/widgets/text/numpad_text.dart]: please remove static variables

File: lib/widgets/text/numpad_text.dart [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+      ),
+      onKeyTap: onKeyTapped,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    width = MediaQuery.of(context).size.width;
+    height = MediaQuery.of(context).size.height;
+    debugPrint('Width :: $width and Height :: $height');
```
Comment (almeynman) [inline lib/widgets/text/numpad_text.dart]: please remove

File: lib/widgets/text/numpad_text.dart [hunk 9]
```diff
@@ -0,0 +1,187 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_ui_kit/color.dart';
+
+class NumPadText extends StatefulWidget {
```
Comment (almeynman) [inline lib/widgets/text/numpad_text.dart]: please make it StatelessWidget

File: lib/widgets/text/numpad_text.dart [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+    debugPrint('Width :: $width and Height :: $height');
+    digitTextStyle = Theme.of(context).textTheme.display2.copyWith(fontFamily: 'CircularPro-Book', fontWeight: FontWeight.normal, color: Colors.black, fontSize: 24.0);
+    return Material(
+        child: Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: <Widget>[
+              Container(
+                alignment: Alignment.bottomRight,
+                color: Colors.white,
+                  height: height/2.5,
```
Comment (almeynman) [inline lib/widgets/text/numpad_text.dart]: height: height

File: lib/widgets/text/numpad_text.dart (hunk 2 above)
Comment (almeynman) [reply lib/widgets/text/numpad_text.dart]: Actually please remove all static and local variables

File: lib/widgets/text/numpad_text.dart [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+  });
+
+  void onKeyTapped(String key) {
+    var result = textEditingController.text;
+    if (key == '.') {
+      if (!result.contains('.')) {
+        result += key;
+      }
+    } else if (key == 'C') {
+      result = result.substring(0, result.length-1);
```
Comment (jayrmotta) [inline lib/widgets/text/numpad_text.dart]: Can you add a test for this feature?

File: lib/widgets/text/numpad_text.dart [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+    } else {
+      result += key;
+    }
+    if (digitsAfterPoint > 0 && result.contains('.')
+        && result.substring(result.indexOf('.')).length > digitsAfterPoint + 1) {
```
Comment (jayrmotta) [inline lib/widgets/text/numpad_text.dart]: Can you add a test for this feature?

File: lib/widgets/text/numpad_text.dart [hunk 13]
```diff
@@ -0,0 +1,143 @@
...
+              Container(
+                alignment: Alignment.bottomRight,
+                color: Colors.white,
+                height: MediaQuery.of(context).size.height * 0.4,
+                width: MediaQuery.of(context).size.width * 0.8,
+                  child: Column(
+                      mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                      children: <Widget>[
+                        Expanded(
+                          child: Row(
```
Comment (jayrmotta) [inline lib/widgets/text/numpad_text.dart]: This row is repeated a couple of time only changing the keys inside.

Can we create a method that receives variable arguments or something like this and produces it?

File: lib/widgets/text/numpad_text.dart [hunk 14]
```diff
@@ -0,0 +1,143 @@
...
+  const KeyItem({@required this.child,this.value,this.onKeyTap});
+
+  @override
+  Widget build(BuildContext context) {
+    assert(debugCheckHasMaterial(context));
+    return Expanded(
+      child: Material(
+        type: MaterialType.transparency,
+        child: InkResponse(
+          radius: MediaQuery.of(context).size.height * 0.05 - 2,
```
Comment (jayrmotta) [inline lib/widgets/text/numpad_text.dart]: What are these magic numbers here?

File: lib/stories/numpads.dart [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+        width: MediaQuery.of(context).size.width,
+        child: SingleChildScrollView(child: _numPadStory(context)));
+  }
+
+  Widget _numPadStory(BuildContext context) {
+    final _textEditingController = TextEditingController();
+    return Column(mainAxisSize: MainAxisSize.min, children: [
+      new Container(
+          child: ConstrainedBox(
+          constraints: const BoxConstraints.tightFor(width: 280.0),
```
Comment (almeynman) [inline lib/stories/numpads.dart]: add propupdater for decimalPlaces

File: lib/widgets/text/numpad_text.dart [hunk 16]
```diff
@@ -0,0 +1,143 @@
...
+import 'package:flutter_ui_kit/color.dart';
+
+class NumPadText extends StatelessWidget {
+
+  final TextEditingController textEditingController;
+  final int digitsAfterPoint;
+
+  const NumPadText({
+    @required this.textEditingController,
+    this.digitsAfterPoint = -1
```
Comment (almeynman) [inline lib/widgets/text/numpad_text.dart]: decimalPlaces

File: lib/widgets/text/numpad_text.dart [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+    if ('0123456789.'.contains(key)) {
+      if (key == '.' && result.contains('.'))
+        return;
+      result += key;
+    } else if (key == 'C') {
+      result = result.substring(0, result.length - 1);
+    } else
+      return;
+
+    if (decimalPlaces != null && result.contains('.')
```
Comment (almeynman) [inline lib/widgets/text/numpad_text.dart]: can you please refactor this into a method

File: lib/widgets/text/numpad_text.dart [hunk 18]
```diff
@@ -0,0 +1,99 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_ui_kit/color.dart';
+import 'package:flutter_ui_kit/text.dart';
+
+class NumPadText extends StatelessWidget {
+
+  final TextEditingController textEditingController;
```
Comment (almeynman) [inline lib/widgets/text/numpad_text.dart]: can we simplify this and use onChange Callback

File: lib/widgets/text/numpad_text.dart (hunk 17 above)
Comment (almeynman) [reply lib/widgets/text/numpad_text.dart]:
```
bool shouldRestrictDecimalPlaces() {
	return decimalPlaces != null && result.contains('.') && result.substring(result.indexOf('.')).length > decimalPlaces + 1;
}
```

File: lib/widgets/text/numpad_text.dart [hunk 19]
```diff
@@ -0,0 +1,99 @@
...
+
+  const NumPadText({
+    @required this.textEditingController,
+    this.decimalPlaces
+  });
+
+  void onKeyTapped(String key) {
+    var result = textEditingController.text;
+    if ('0123456789.'.contains(key)) {
+      if (key == '.' && result.contains('.'))
```
Comment (almeynman) [inline lib/widgets/text/numpad_text.dart]: `alreadyHasADot`

File: lib/widgets/text/numpad_text.dart (hunk 18 above)
Comment (almeynman) [reply lib/widgets/text/numpad_text.dart]: `typedef Callback = void Function(String value)`

`final Callback onChange;`

File: lib/widgets/text/numpad_text.dart [hunk 20]
```diff
@@ -0,0 +1,99 @@
...
+      result = result.substring(0, result.length - 1);
+    }
+    textEditingController.text = result;
+  }
+
+  KeyItem buildKeyItem(String val) {
+    return KeyItem(
+        value: val ,
+        child: (val != 'C')
+            ? Text(val , textAlign: TextAlign.center , style: AppText.body4)
```
Comment (almeynman) [inline lib/widgets/text/numpad_text.dart]: `numPadTextStyle`

File: test/widgets/text/numpad_text_test.dart [hunk 21]
```diff
@@ -0,0 +1,46 @@
...
+      await tester.pumpWidget(wrapInMaterialApp(
+          NumPadText(textEditingController: _textEditingController)));
+      expect(find.widgetWithIcon(NumPadText, Icons.arrow_back), findsOneWidget);
+    });
+  });
+
+  group('NumPadText input', () {
+    final _textEditingController = TextEditingController();
+
+    testWidgets('on tap key tests', (WidgetTester tester) async {
```
Comment (almeynman) [inline test/widgets/text/numpad_text_test.dart]: please add onChange test

File: lib/widgets/text/numpad_text.dart [hunk 22]
```diff
@@ -0,0 +1,108 @@
...
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+      children: [
+        Container(
+          child: buildRow(['1', '2', '3'])
```
Comment (jayrmotta) [inline lib/widgets/text/numpad_text.dart]: This looks neat!! Congrats 🚀

**Pull request merged**